src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/VersionService.java

```diff
@@ -28,24 +28,37 @@
 import ee.ria.xroad.common.Version;
 
 import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.openapi.model.VersionInfo;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 /**
  * service class for handling X-Road version information
  */
 @Slf4j
 @Service
-@Transactional
 @PreAuthorize("isAuthenticated()")
 public class VersionService {
+    public static final int MIN_SUPPORTED_JAVA_VERSION = 8;
+    public static final int MAX_SUPPORTED_JAVA_VERSION = 11;
+
 
     /**
-     * Returns X-Road software version number
+     * Returns X-Road software version number and java version information
      * @return
      */
-    public String  getVersion() {
-        return Version.XROAD_VERSION;
+    public VersionInfo getVersionInfo() {
```

Comment:
Using an OpenAPI Model Object in the service layer – instead, a separate DTO should be used and converted in the controller layer.

To have a clear separation between the service and controller layers, it was decided to avoid using OpenAPI models in the service layer. More info about the architectural decisions here: https://confluence.niis.org/display/XRDDEV/API+backend+implementation

Reply:
I definitely and absolutely do not agree with this "architectural" decision but will make the change.